parser.go

```diff
@@ -16,6 +17,62 @@ type SSHHost struct {
 	ProxyCommand      string
 	HostKeyAlgorithms string
 	IdentityFile      string
+	LocalForwards     []Forward
+	RemoteForwards    []Forward
+	DynamicForwards   []DForward
+}
+
+// Forward defines a single port forward entry
+type Forward struct {
+	InHost  string
+	InPort  int
+	OutHost string
+	OutPort int
+}
+
+// NewForward returns Forward object parsed from LocalForward or RemoteForward string
+func NewForward(f string) (Forward, error) {
+	r := regexp.MustCompile(`((\S+):)?(\d+)\s+(\S+):(\d+)`)
+	m := r.FindStringSubmatch(f)
+
+	InPort, err := strconv.Atoi(m[3])
+	if err != nil {
+		return Forward{}, err
+	}
+
+	OutPort, err := strconv.Atoi(m[5])
+	if err != nil {
+		return Forward{}, err
+	}
+
+	return Forward{
+		InHost:  m[2],
+		InPort:  InPort,
+		OutHost: m[4],
+		OutPort: OutPort,
+	}, nil
+}
+
+// DForward defines a single dynamic port forward entry
+type DForward struct {
```

Comment:
nit: Let's spell out the full name i.e. `DynamicForward`.

Reply:
Ok, I will change that. Working on some test improvements too and fixed some linting issues in the tests.

Reply:
Done! Also, maybe I do not understand your lexer well enough, but in the parse loop you check `if next.typ != itemValue` a lot. I cannot think of a scenario this would be `true`. So I removed those checks in my addition.

Reply:
The lexer is inspired by the one used in text/template of the standard library: https://golang.org/src/text/template/parse/lex.go i.e. it doesn't just regex but just emits the tokens and then the parser checks if those are as expected or not. In your addition you use regex and do the error checking there on the parser side.
The initial lexer was mostly for learning purposes so I'm find with your additions even though the approach is slightly different.